runners/gradle-plugin/src/main/kotlin/org/jetbrains/dokka/gradle/GradleDokkaSourceSetBuilder.kt

```diff
@@ -156,8 +155,9 @@ open class GradleDokkaSourceSetBuilder(
         sourceRoot(project.file(path))
     }
 
+    @Suppress("DEPRECATION") // TODO [beresnev] ConfigureUtil will be removed in Gradle 8
     fun sourceLink(c: Closure<in GradleSourceLinkBuilder>) {
-        val configured = ConfigureUtil.configure(c, GradleSourceLinkBuilder(project))
+        val configured = org.gradle.util.ConfigureUtil.configure(c, GradleSourceLinkBuilder(project))
         sourceLinks.add(configured)
     }
```

Comment:
`ConfigureUtil` is going to be removed in Gradle 8 as stated in the javadoc. Suggested replacement is migrating to `Action<in GradleSourceLinkBuilder>`, and we have these methods, but they don't work for groovy-based scripts.

I've also investigated it and it looks like everything should work (everything is done the same way as in documentation), but it just doesn't -- fails with some strange error. 

Worst case scenario is we'll need to solve this when migrating to gradle 8, but I think we'll get to it sooner when we start integrating dokka into the kotlin gradle plugin.